arcaflow_plugin_horreum_client/horreum_client_plugin.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+import sys
+import typing
+import requests
+import json
+
+from arcaflow_plugin_sdk import plugin
+from horreum_client_schema import InputParams, SuccessOutput, ErrorOutput
+
+
+@plugin.step(
+    id="horreum-client",
+    name="Horreum Client",
+    description="Uploads an object to the Horreum server",
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def horreum_client(
+    params: InputParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    auth_url = (
+        f"{params.horreum_keycloak_url}/realms/horreum/protocol/openid-connect/token"
+    )
```

Comment:
Does Horreum support the concept of a persistent client API Key as opposed to having your Arca client actually log in with username/password?

Reply:
The feedback I got is that persistent client API keys are partially supported, but the current method is going to change soon. So I think we will revisit this later.